### ISSUE TITLE
"Error: Key cannot be the null string" on ignoble BN ebooks issue ?

### ISSUE THREAD

**Issue by lopesta** [apprenticeharper/DeDRM_tools#383]
![null](https://user-images.githubusercontent.com/24236821/31607410-dffb9ff6-b274-11e7-9de0-020b564e1712.gif)
I installed "ActiveState ActivePython 2.7.2.5 (32-bit) & then "Python 2.7 pycrypto-2.1.0"

i was able to decrypt adobe pdf files correctly and adobe epub too ,
was getting for normal adobe epub an error message saying "Error: File name in directory "OEBPS/html/grap...etc " then i fixed with first epub fixer apply on source drmed file then output is drm fixed file then decoding succefully from drm fixed into no-drm fixed .. this was for some epub files .

now onto barnes and noble .. i have no success with decrypting/decoding the epub file

it say "Error: Key cannot be the null string" on ignobleepub_v3.5 ...

i had put the account name as i have used carefully entact , also i put my full cc # used into ignoblekeygen_v2.4   with no success ..

i tried also downloading nook study 2014 (i think from epubsoft.com) so using the old mechanism without any success .. also tried renaming files before decoding still error arise

what i am missing here please ?? do i need to use old version of active state active python and/or Pycrypto ???

bottom line ..

everything working in adobe pdf decoding except i didn't tested BN pdf files ..
and adobe epub also working successfully except BN epubs .

for account name i used firstname lastname as were it used in the order and the same cc ..

any help will be much appreciated

**Comment by lopesta** [comment]
Sorry i think i solved the matter completely .. by using the latest (this drm removal package) ..
i used my my email and password of bn to retrieve the key then decoded them succefully ..
